test/FunctionalTests/Balancer/ConnectionTests.cs

```diff
@@ -554,7 +554,7 @@ public async Task SocketSendsBytes_BeforeCall_LargeData_Error()
             () => Logs.Count(l => l.EventId.Name == "CheckingSocket") >= RequiredCheckLogCount,
             "Wait for multiple checking socket logs.").DefaultTimeout();
 
-        Assert.True(Logs.Any(l => l.EventId.Name == "ErrorCheckingSocket" && l.Exception?.Message == "The server sent 65536 bytes to the client before a connection was established. This exceeds maximum data allow."), "Socket was unexpectedly disconnected with a bad state.");
+        Assert.True(Logs.Any(l => l.EventId.Name == "ErrorCheckingSocket" && l.Exception is { } ex && ex.Message.Contains("Maximum allowed data exceeded.")), "Socket was disconnected because of large data.");
```

Comment:
string based match seems ... sloppy here; I wonder if a specific `InvalidOperationException` subclass might be preferable

Reply:
It's not perfect, but it works. Creating a new exception type to identify every place an error is thrown doesn't scale.